### PR TITLE
Fix YouTube plugin throwing errors on calling player methods (close #1043)

### LINKED ISSUE #1043: Fix YouTube plugin throwing errors on calling player methods
**Describe the bug**
The YouTube plugin can throw errors if player methods are run before the player instance is ready.

The event listeners can be attached in the player `onReady` callback to prevent this.

### PR DESCRIPTION
**Pull request by greg-el**
- #1043

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://docs.snowplowanalytics.com/docs/contributing/contributor-license-agreement/ to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

File: plugins/browser-plugin-youtube-tracking/src/index.ts [hunk 1]
```diff
@@ -107,22 +107,40 @@ function handleYouTubeIframeAPI() {
...
-      addListeners(trackingQueue.pop()!);
+      initialisePlayer(trackingQueue.pop()!);
     }
   }
 }
 
-function addListeners(conf: TrackingOptions) {
+function initialisePlayer(conf: TrackingOptions) {
+  trackedPlayers[conf.mediaId] = {
+    player: new YT.Player(conf.mediaId, { events: { onReady: (e: any) => playerReady(e, conf) } }),
```
Comment (paulboocock) [inline plugins/browser-plugin-youtube-tracking/src/index.ts:127]: nit: `any`

File: plugins/browser-plugin-youtube-tracking/src/index.ts [hunk 2]
```diff
@@ -107,22 +107,40 @@ function handleYouTubeIframeAPI() {
...
+      enabled: false,
+    },
+    volumeTracking: {
+      prevVolume: 0,
+      enabled: false,
+    },
+  };
+}
+
+function playerReady(event: any, conf: TrackingOptions) {
```
Comment (paulboocock) [inline plugins/browser-plugin-youtube-tracking/src/index.ts:140]:
```suggestion
function playerReady(event: any, conf: TrackingOptions) {
```

nit: `any`

File: plugins/browser-plugin-youtube-tracking/src/index.ts [hunk 3]
```diff
@@ -137,23 +155,9 @@ function addListeners(conf: TrackingOptions) {
       youtubeEvent(trackedPlayers[conf.mediaId].player, YTEvent.PLAYBACKRATECHANGE, conf),
   };
 
-  const playerEvents: Record<string, Function> = {};
   conf.youtubeEvents.forEach((e) => {
-    playerEvents[e] = builtInEvents[e];
+    player.addEventListener(e, (f) => builtInEvents[String(e)](f));
```
Comment (paulboocock) [inline plugins/browser-plugin-youtube-tracking/src/index.ts:159]: Can we expand some of the variable names here to be more descriptive?

**Review by paulboocock — APPROVED** [review]
Couple of nits, looks good though.

**Pull request merged**